staging/src/k8s.io/kubectl/pkg/cmd/debug/debug.go

```diff
@@ -590,9 +610,37 @@ func (o *DebugOptions) generatePodCopyWithDebugContainer(pod *corev1.Pod) (*core
 	c.Stdin = o.Interactive
 	c.TTY = o.TTY
 
+	if len(o.Overrides) > 0 {
+		mergedContainer, err := mergeContainer(c, o.Overrides)
+		if err != nil {
+			return nil, "", fmt.Errorf("failed to merge container from overrides :%v", err)
+		}
+		*c = *mergedContainer
+	}
 	return copied, name, nil
 }
 
+func mergeContainer(container *corev1.Container, overRides string) (*corev1.Container, error) {
+	if len(overRides) == 0 {
+		return container, nil
+	}
+	target, err := json.Marshal(container)
```

Comment:
This is not right, we should be laveraging our internal codec mechanism, compare with https://github.com/kubernetes/kubernetes/blob/19c7089245268e59e46b0a19da9fed45d18cd73f/staging/src/k8s.io/kubectl/pkg/cmd/run/run.go#L624-L630 which has that same functionality.